sdks/go/pkg/beam/io/filesystem/memfs/memory.go

```diff
@@ -54,14 +55,18 @@ func (f *fs) List(_ context.Context, glob string) ([]string, error) {
 	f.mu.Lock()
 	defer f.mu.Unlock()
 
-	pattern, err := regexp.Compile(glob)
-	if err != nil {
-		return nil, err
+	globNoScheme := strings.TrimPrefix(glob, "memfs://")
+	if globNoScheme == glob {
+		return nil, fmt.Errorf("invalid pattern %q does not have prefix memfs://", glob)
 	}
 
 	var ret []string
 	for k := range f.m {
-		if pattern.MatchString(k) {
+		matched, err := path.Match(globNoScheme, strings.TrimPrefix(k, "memfs://"))
```

Comment:
Please switch this to (path/filepath) filepath.Match instead, for consistency with the gcs and local implementations. In practice it's always linux.  Please also update the documentation in filesystem.Interface.List to be explicit as you suggested.

Thank you for the contribution!

Reply:
Done.

Please note that I also made the pattern accepted by List() more tolerant. It no longer requires a memfs:// prefix.